### ISSUE TITLE
Syntaxnet problem

### ISSUE THREAD

**Issue by inferrna** [bazelbuild/bazel#2374]
Can you please look at https://github.com/tensorflow/models/issues/902 ? The error message is `error loading package '@jpeg//': Extension file not found. Unable to load package for '//third_party:common.bzl': BUILD file not found on package path and referenced by '@org_tensorflow//tensorflow/core/platform/default/build_config:jpeg'.`
Seems like it's not a bazel problem, but very related. May be you can help to solve it, so tensorflow & syntaxnet developers still can't. Steps to reproduce:
```sh
git clone https://github.com/tensorflow/models.git
cd models/syntaxnet/
git submodule update --init --recursive
cd tensorflow
./configure #All to default except C/C++ compilers - mustn't be empty
cd ../
bazel test syntaxnet/... util/utf8/...
```
thank you.

**Comment by Rajendrasinh-AI** [comment]
is there any previous commit that works perfectly fine?

**Comment by inferrna** [comment]
Yes, dee751fafb66e511c6fec02b572670b50bc517fa is working, but needs some prepare.
1. Downgrade bazel to 3.2
2. Apply the patch below
```patch
diff --git a/tensorflow/workspace.bzl b/tensorflow/workspace.bzl
index c20cee7..21e3527 100644
--- a/tensorflow/workspace.bzl
+++ b/tensorflow/workspace.bzl
@@ -225,7 +225,7 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
 
   native.new_http_archive(
     name = "zlib_archive",
-    url = "http://zlib.net/zlib-1.2.8.tar.gz",
+    url = "http://bazel-mirror.storage.googleapis.com/zlib.net/zlib-1.2.8.tar.gz",
     sha256 = "36658cb768a54c1d4dec43c3116c27ed893e88b02ecfcb44f2166f9c0b7f2a0d",
     strip_prefix = "zlib-1.2.8",
     build_file = str(Label("//:zlib.BUILD")),
```

**Comment by kyamama47** [comment]
Thank you!! It worked well on Ubuntu 14.04.

**Comment by negacy** [comment]
I downgraded bazel to 3.2 but I was not able to build it on OSX. I don't understand the patch mentioned though. What are the directories `a/tensorflow/workspace.bzl` and `b/tensorflow/workspace.bzl`

**Comment by kyamama47** [comment]
I did the following    

    git clone --recursive https://github.com/tensorflow/models.git
    cd models
    git checkout dee751fafb66e511c6fec02b572670b50bc517fa
    cd syntaxnet
    git submodule update --init --recursive
    cd tensorflow  # place  diff.patch.txt  here and remove (.txt)   
    patch -p1 < diff.patch 
    ./configure
    cd ..
    bazel test syntaxnet/... util/utf8/...

[diff.patch.txt](https://github.com/bazelbuild/bazel/files/716847/diff.patch.txt)

Hope that helps.

**Comment by negacy** [comment]
@kyamama47: following your steps, this is what I got:
➜  tensorflow git:(8ed0023) ✗ ./configure
~/git/models/syntaxnet/tensorflow ~/git/models/syntaxnet/tensorflow
Please specify the location of python. [Default is /usr/bin/python]:
Do you wish to build TensorFlow with Google Cloud Platform support? [y/N] N
No Google Cloud Platform support will be enabled for TensorFlow
Do you wish to build TensorFlow with Hadoop File System support? [y/N] N
No Hadoop File System support will be enabled for TensorFlow
Can't find swig.  Ensure swig is in $PATH or set $SWIG_PATH.

**Comment by kyamama47** [comment]
I think swig is not installed in your PC.
In the case of Ubuntu, 

    sudo apt-get install swig 

But I'm not sure how install swig to OSX. 
Could you search for 'install swig mac'.

**Comment by negacy** [comment]
Right, I installed swig but it raising another error.
..............
Unhandled exception thrown during build; message: Unrecoverable error while evaluating node 'WORKSPACE_FILE:[/Users/negacy/git/models/syntaxnet]/[WORKSPACE], 2' (requested by nodes 'EXTERNAL_PACKAGE:[/Users/negacy/git/models/syntaxnet]/[WORKSPACE]')
INFO: Elapsed time: 1.135s
java.lang.RuntimeException: Unrecoverable error while evaluating node 'WORKSPACE_FILE:[/Users/negacy/git/models/syntaxnet]/[WORKSPACE], 2' (requested by nodes 'EXTERNAL_PACKAGE:[/Users/negacy/git/models/syntaxnet]/[WORKSPACE]')
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:447)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:496)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: Multiple entries with same key: @org_tensorflow//tensorflow:workspace.bzl=com.google.devtools.build.lib.syntax.Environment$Extension@22275bc4 and @org_tensorflow//tensorflow:workspace.bzl=com.google.devtools.build.lib.syntax.Environment$Extension@22275bc4
	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:136)
	at com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:98)
	at com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:84)
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:295)
	at com.google.devtools.build.lib.packages.WorkspaceFactory.execute(WorkspaceFactory.java:188)
	at com.google.devtools.build.lib.packages.WorkspaceFactory.execute(WorkspaceFactory.java:173)
	at com.google.devtools.build.lib.skyframe.WorkspaceFileFunction.compute(WorkspaceFileFunction.java:112)
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:388)
	... 4 more
java.lang.RuntimeException: Unrecoverable error while evaluating node 'WORKSPACE_FILE:[/Users/negacy/git/models/syntaxnet]/[WORKSPACE], 2' (requested by nodes 'EXTERNAL_PACKAGE:[/Users/negacy/git/models/syntaxnet]/[WORKSPACE]')
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:447)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:496)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: Multiple entries with same key: @org_tensorflow//tensorflow:workspace.bzl=com.google.devtools.build.lib.syntax.Environment$Extension@22275bc4 and @org_tensorflow//tensorflow:workspace.bzl=com.google.devtools.build.lib.syntax.Environment$Extension@22275bc4
	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:136)
	at com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:98)
	at com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:84)
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:295)
	at com.google.devtools.build.lib.packages.WorkspaceFactory.execute(WorkspaceFactory.java:188)
	at com.google.devtools.build.lib.packages.WorkspaceFactory.execute(WorkspaceFactory.java:173)
	at com.google.devtools.build.lib.skyframe.WorkspaceFileFunction.compute(WorkspaceFileFunction.java:112)
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:388)
	... 4 more

**Comment by kyamama47** [comment]
I have never seen that error before.
I'm afraid i can't help you.

**Comment by weimate** [comment]
Using these steps: https://github.com/bazelbuild/bazel/issues/2374#issuecomment-273794464, i have the following error at `bazel test syntaxnet/... util/utf8/...`:

```
INFO: Found 68 targets and 17 test targets...
ERROR: /root/.cache/bazel/_bazel_root/bb733f80bf7661ee0ae426cddd50a86c/external/com_googlesource_code_re2/BUILD:11:1: C++ compilation of rule '@com_googlesource_code_re2//:re2' failed: gcc failed: error executing command /bin/gcc -U_FORTIFY_SOURCE '-D_FORTIFY_SOURCE=1' -fstack-protector -Wall -Wl,-z,-relro,-z,now -B/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-canonical-system-headers ... (remaining 32 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
external/com_googlesource_code_re2/re2/dfa.cc: In constructor 're2::DFA::State::State()':
external/com_googlesource_code_re2/re2/dfa.cc:117:10: error: unknown array size in delete
   struct State {
          ^~~~~
external/com_googlesource_code_re2/re2/dfa.cc: In member function 're2::DFA::State* re2::DFA::CachedState(int*, int, uint32_t)':
external/com_googlesource_code_re2/re2/dfa.cc:719:9: note: synthesized method 're2::DFA::State::State()' first required here 
   State state;
         ^~~~~
INFO: Elapsed time: 50.976s, Critical Path: 23.66s
```

but at least not failing on `@jpeg://` module. Any suggestions? Thanks!

**Comment by damienmg** [comment]
@negacy: looks like you hitted a genuine bug, which version of bazel are you using?

For the other, I don't see issue that are Bazel related in the questions and I see that Gunhan is already helping you on the TF tracker :)

**Comment by damienmg** [comment]
And for @negacy, this is a duplicate of #1469 which is fixed with bazel 0.4.3. Since you got help from gunhan on the TF tracker, I am closing this issue.
